hoomd/Trigger.h

```diff
@@ -27,10 +27,26 @@ class PYBIND11_EXPORT Trigger
          *  @param timestep Time step to query
          *  @returns `true` if the operation should occur, `false` if not
         */
-        virtual bool operator()(uint64_t timestep)
+        bool operator()(uint64_t timestep)
             {
-            return false;
+            if (m_last_timestep == timestep)
+                {
+                return m_last_trigger;
+                }
+            else
+                {
+                auto triggered = compute(timestep);
+                m_last_timestep = timestep;
+                m_last_trigger = triggered;
+                return triggered;
+                }
             }
+
+        virtual bool compute(uint64_t timestep) = 0;
+
+    private:
+            uint64_t m_last_timestep;
+            bool m_last_trigger;
```

Comment:
hmm, it makes sense, but I am not sure if it is in scope for now.

Reply:
We could consider that for a later PR that updates all uses of `unsigned int timestep`. For now, we are implementing new functionality with `uint64_t` and there is no loss of generality because the calling functions that use `unsigned int` will have their values automatically converted up to `uint64_t` when they call a new method.